### PR TITLE
feature: cabins 2.0 (dok-166, dok-162, dok-160)

### PR DESCRIPTION
**Pull request by hermanholmoy**
#### What problem/feature does this pull request fix/add?
New booking flow. The flow is as follows:
1. Choose check in and check out date, as well as which cabin to book.
2. Input contact info and number of internal/external participants.
3. Accept (or decline) the contract.
4. Review a summary of the booking, click send.
5. A mail is sent to Hyttestyret, prompting them to either accept or decline the booking. The user gets a summary mail.
6. When a booking is accepted or declined, the user gets a confirmation mail, and the booking is created.

#### (if not a new problem) How was this problem solved earlier?
Old version had two pages, one for check in/out selection and another for contact info/contract. This solution gave little feedback to the user on how far in the booking process he/she has come. The new version uses a stepper which shows what step the user is on (see screenshot below). 

#### How did I fix this problem?
New stepper component with clear feedback on what the status of the current booking is.

### Checklist

- [ ] All tests have passed
- [x] I have created new tests for this feature (may not be applicable to all pull requests)
- [x] I am pleased with the readability of the code (if not, state where you need input)

### Stepper on the check in/out step
Screenshot 2021-09-10 at 13 11 25

### CODE REVIEW

**Review by larwaa — CHANGES_REQUESTED** [review]
This is starting to look really great, just some final polishing then I think we can make this live! 🤩 Massive props for an absolutely amazing effort in refactoring some of the components, and generally increasing the code quality even further! 🚀 I hope you're not discouraged by the amount of comments, it's just a really long PR! The quality really is 👌

File: backend/apps/cabins/models.py [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
-    bookTo = models.DateField()
-    price = models.IntegerField()
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    phone = models.CharField(max_length=8)
+    receiver_email = models.EmailField(max_length=100)
+    check_in = models.DateField()
+    check_out = models.DateField()
     cabins = models.ManyToManyField(Cabin)
+    timestamp = models.DateTimeField()
```
[the remark this answers is not in the document]
Comment (hermanholmoy) [reply backend/apps/cabins/models.py:27]: Does this make `booking.timestamp = timezone.now()` (in cabins/mutations.py) redundant? Also, I'm assuming you mean `auto_now_add`? https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.DateField.auto_now_add

[the remark this answers is not in the document]
Comment (larwaa) [reply backend/apps/cabins/models.py:27]: Yes and yes 🙈

File: frontend/src/pages/cabins/index.tsx [hunk 2]
```diff
@@ -126,9 +155,18 @@ const CabinsPage: NextPage = () => {
...
+                <Button
+                  variant="contained"
+                  endIcon={<NavigateNextIcon />}
+                  disabled={!isLoggedIn}
+                  classes={{ root: classes.button }}
+                >
+                  Book nå
+                </Button>
+              </Tooltip>
+            </PermissionRequired>
```
Comment (larwaa) [inline frontend/src/pages/cabins/index.tsx:169]: Keep in mind that `<PermissionRequired />` will return `null` in all cases where the user does not have the required permission, which will necessarily be the case if the user is not logged in. Either move the `<Tooltip />` outside `<PermissionRequired />`, or expand the `<PermissionRequired />` component to accept a fallback-component. See DOK-216 for more information.

**Review by larwaa — APPROVED** [review]
LGTM! 🥳🚀 Can't wait to see this go live! Great job, @hermanholmoy, @sverress, @jorul! A big and important addition to the website, and a gold standard for how to properly test features! ⭐️

**Pull request merged**